SharedCode/Database/DatabaseManager.cs

```diff
@@ -96,5 +98,19 @@ public Result<List<T>> GetAllData<T>() where T : new()
                 return Result.Fail<List<T>>("Could not get all data");
             }
         }
+
+        public Result<T> GetDataById<T>(int id, string tableName) where T : new()
+        {
+            try
+            {
+                var query = "SELECT * FROM " + tableName + " WHERE Id LIKE ?";
```

Comment:
Better use query builder for Sqlite. Refer to the documentation you shared.

Reply:
ok i'll take it a look

Reply:
fixed